### PR TITLE
Show infotext with description for item entities

### PR DESCRIPTION
**Pull request by RealBadAngel**
screenshot_20160108_155704
screenshot_20160108_155806

### CODE REVIEW

File: src/content_cao.cpp [hunk 1]
```diff
@@ -1850,5 +1850,10 @@ std::string GenericCAO::debugInfoText()
 	return os.str();
 }
 
+std::string GenericCAO::infoText()
```
Comment (nerzhul) [inline src/content_cao.cpp]: no, set this definition to content_cao.h directly to permit inlining implictly by compiler.

Comment (nerzhul) [inline src/content_cao.cpp]: Also, return const std::string instead of std::string

File: src/content_cao.h [hunk 2]
```diff
@@ -201,6 +201,8 @@ class GenericCAO : public ClientActiveObject
 			float time_from_last_punch=1000000);
 
 	std::string debugInfoText();
+	
+	std::string infoText();
```
Comment (nerzhul) [inline src/content_cao.h]: don't forget virtual keyword because it's also in ClientActiveObject

**Comment by PilzAdam** [comment]
Can you document the new field in lua_api.txt?

File: src/content_cao.h [hunk 3]
```diff
@@ -201,6 +201,8 @@ class GenericCAO : public ClientActiveObject
 			float time_from_last_punch=1000000);
 
 	std::string debugInfoText();
```
Comment (nerzhul) [inline src/content_cao.h]: there is a bugfix to do there. debugInfoText is virtual in parent class. Please set it virtual too

**Comment by RealBadAngel** [comment]
@PilzAdam done

**Comment by sfan5** [comment]
:+1:

**Comment by nerzhul** [comment]
@PilzAdam remove bad suggestions about virtual operator :)

**Comment by paramat** [comment]
I'd like to give this a +1 but there's no description, is the top screenshot the before or the after?
There's no description of the optional infotext.

**Comment by HybridDog** [comment]
Why is the infotext shown in the top left part of the minetest window and not next to the + cursor?

**Comment by Amaz1** [comment]
> ... is the top screenshot the before or the after? 

I don't think that either are the before/after screenshots. From the small bit of testing that I've done, it seems that the top one is how it looks without debug info, and the bottom one is with debug info on.

>How does this change the information shown?

It makes the info in debug mode (hp & armor groups) appear a line further down the screen, and (somewhat obviously!) shows the "infotext" of the item being pointed at.

**Comment by RealBadAngel** [comment]
@HybridDog infotext is shown at the same position like for nodes, its fixed.

**Comment by paramat** [comment]
:+1:

**Comment by sofar** [comment]
Tested this and it's great, simple implementation as well, so :+1: from me.

**Comment by nerzhul** [comment]
@RealBadAngel i'm okay with your PR except the inlining comments

**Comment by RealBadAngel** [comment]
@nerzhul ah, forgot bout that, will do that later today

**Comment by sfan5** [comment]
:+1:

**Comment by paramat** [comment]
Will merge later.

**Comment by paramat** [comment]
Nerzhul's issue is fixed now? If so this can be merged.

**Comment by paramat** [comment]
87291ea44a61614d5bd9efc0657926da3f867b5f

**Pull request closed without merging**